node/node_newblock.go

```diff
@@ -88,14 +89,16 @@ func (node *Node) proposeNewBlock() (*types.Block, error) {
 
 	// TODO: integrate staking transaction into tx pool
 	pendingStakingTransactions := types2.StakingTransactions{}
-	node.pendingStakingTxMutex.Lock()
-	for _, tx := range node.pendingStakingTransactions {
-		pendingStakingTransactions = append(pendingStakingTransactions, tx)
+	// Only process staking transactions after pre-staking epoch happened.
+	if node.Blockchain().Config().IsPreStaking(node.Worker.GetNewEpoch()) {
```

Comment:
this will make testing annoying for rest of team - need to let them know only happens after epoch 3 on debug.sh

Reply:
In test/local net, PreStakingEpoch = big.NewInt(0). There is no issue for testing.

Reply:
I meant 3 is slow for testing today, if staking was at epoch 1 , would iterate faster , no?